geopandas/tools/overlay.py

```diff
@@ -14,12 +15,19 @@ def _ensure_geometry_column(df):
     If another column with that name exists, it will be dropped.
     """
     if not df._geometry_column_name == "geometry":
```

Comment:
(We might want to check why the code for symm diff actually requires this and couldn't be updated to be more robust regarding geometry column name, though)